python/ray/data/tests/test_dataset.py

```diff
@@ -4688,6 +4689,30 @@ def get_node_id():
     assert set(locations) == {node1_id, node2_id}
 
 
+def test_stats_actor_cap_num_stats(ray_start_cluster):
+    actor = _StatsActor.remote(3)
+    metadatas = []
+    task_idx = 0
+    for uuid in range(3):
+        metadatas.append(
+            BlockMetadata(
+                num_rows=uuid,
+                size_bytes=None,
+                schema=None,
+                input_files=None,
+                exec_stats=None,
+            )
+        )
+        actor.record_start.remote(uuid)
+        actor.record_task.remote(uuid, task_idx, metadatas[-1])
+    for uuid in range(3):
+        assert ray.get(actor.get.remote(uuid))[0][task_idx] == metadatas[uuid]
+    # Add the fourth stats to exceed the limit.
+    actor.record_start.remote(3)
+    # The first stats (with uuid=0) should have been purged.
+    assert ray.get(actor.get.remote(0))[0] == {}
```

Comment:
An even stronger test would be to query the internal dict sizes, to verify the deletion actually happened.

Reply:
Added a private method to the StatsActor to query the dict sizes.